Explain how to disable app config reload in ASP.NET Core 5.0 (#17386)
dotnet/AspNetCore.Docs: aspnetcore/fundamentals/host/generic-host.md

If the timeout period expires before all of the hosted services stop, any remaining active services are stopped when the app shuts down. The services stop even if they haven't finished processing. If services require additional time to stop, increase the timeout.

**Key**: `shutdownTimeoutSeconds`  
**Type**: `int`  
**Default**: 5 seconds  
**Environment variable**: `<PREFIX_>SHUTDOWNTIMEOUTSECONDS`

To set this value, use the environment variable or configure `HostOptions`. The following example sets the timeout to 20 seconds:

[!code-csharp[](generic-host/samples-snapshot/3.x/Program.cs?name=snippet_HostOptions)]

### Disable app configuration reload on change

> By default, an app configuration file such as *appsettings.json* is reloaded when one of its property values changes. To disable this reload behavior in ASP.NET Core 5.0 Preview 3 or later, set the `hostBuilder:reloadConfigOnChange` key to `false`.

Reviewer: ```suggestion
By default, an app configuration file such as *appsettings.json* is reloaded when the file changes. To disable this reload behavior in ASP.NET Core 5.0 Preview 3 or later, set the `hostBuilder:reloadConfigOnChange` key to `false`.
```

Are we really going to specify preview versions in the docs? Shouldn't this whole new section only show up in the 5.0 docs?

Author: I think it's important to call out the preview number because preview 3 isn't available yet. Hang tight, and I'll make this appear for 5.0+ only.

If the timeout period expires before all of the hosted services stop, any remaining active services are stopped when the app shuts down. The services stop even if they haven't finished processing. If services require additional time to stop, increase the timeout.

**Key**: `shutdownTimeoutSeconds`  
**Type**: `int`  
**Default**: 5 seconds  
**Environment variable**: `<PREFIX_>SHUTDOWNTIMEOUTSECONDS`

To set this value, use the environment variable or configure `HostOptions`. The following example sets the timeout to 20 seconds:

[!code-csharp[](generic-host/samples-snapshot/3.x/Program.cs?name=snippet_HostOptions)]

### Disable app configuration reload on change

> By default, an app configuration file such as *appsettings.json* is reloaded when the file changes. To disable this reload behavior in ASP.NET Core 5.0 Preview 3 or later, set the `hostBuilder:reloadConfigOnChange` key to `false`.

Reviewer 2: ```suggestion
By [default](xref:fundamentals/configuration/index#default), *appsettings.json* and *appsettings.*`Environment`*.json* are reloaded when the file changes. To disable this reload behavior in ASP.NET Core 5.0 Preview 3 or later, set the `hostBuilder:reloadConfigOnChange` key to `false`.
```

Reviewer 2: See https://review.docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-3.1&branch=pr-en-us-17152#default-configuration which should merge soon.
